samwise.js

```diff
@@ -1,75 +1,74 @@
 /**
```

Comment:
SCRIPT-This logic might be clearer if you used a switch statement instead of multiple if-else statements.